### ISSUE TITLE
Serializing Material Trips on Reference to "LinearGradient"

### ISSUE THREAD

**Issue by corfidbizna** [SirWaddles/JohnWickParse#19]
I'm attempting to serialize a .uexp and .uasset pair forming a shader from Kingdom Hearts 3, but I hit this error: 
```
Error: Property error occurred:
Export type: Material
Property Tag: MaterialAttributes (StructProperty) 36
Struct Type: MaterialAttributesInput
Property Tag: /Engine/Functions/Engine_MaterialFunctions01/Gradient/LinearGradient (/Engine/Functions/Engine_MaterialFunctions01/Gradient/LinearGradient) 0
Could not read property type: /Engine/Functions/Engine_MaterialFunctions01/Gradient/LinearGradient at pos 197
```

I've got the LinearGradient's .uexp and .uasset position in the directory hierarchy correct relative to the file I'm trying to serialize and even have the john-wick-parse.exe in the root directory in case it's looking relative to itself, but to no avail. I might just be using the tool wrong, but I can't find anything concerning what to do differently from searching around. I don't think it's anything specific to KH3 that's breaking since LinearGradient is a fairly standard material function, but I'm not entirely sure.

**Comment by SirWaddles** [comment]
I'm not sure sorry, but it looks like there might be a struct somewhere along the way that's not getting read properly. Everything after `MaterialAttributesInput` I believe is being parsed incorrectly.

Unfortunately, even if I was able to find what the problem was, Fortnite no longer uses this format, and it's not supported by JWP any more (newer versions won't read uexp files.)

**Comment by corfidbizna** [comment]
Thank you for the quick reply! Does that mean there's a new / different process for reading UE4 materials using JWP or should I start investigating other means to get at the parameters?

**Comment by SirWaddles** [comment]
The new format applies to all assets (not just materials), and is called "unversioned properties". I think it was added in UE 4.26, though I'm not certain. I believe it's a feature that is opt-in, so if a game upgrades to 4.26, it doesn't necessarily mean they'll be using the new format.

JWP is only designed to be able to parse files from Fortnite, and the new format only makes the assets more tightly bound to the game they're from.

Basically I'm saying that this is likely an issue with the parser, and it will never be fixed (unless you want to fork from an earlier version of the repo) as it's not a supported use case.

I'm sorry I can't be of more help, I suppose it's time to look for alternatives.

**Comment by corfidbizna** [comment]
Ah, alright then. I might end up forking from an earlier version like you say since starting from anything will be better than starting from nothing. Thanks for the help regardless. I wish you luck with the future shenanigans Fortnite may throw at you. : )
